### DISCUSSION TITLE
Make the "withdrawMultiple" function work with multiple recipients

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#773] [General]
The `withdrawMultiple` function is severely limited at the moment: it only works if all the streams in the input array have the same recipient. This issue has been previously flagged by @razgraf [here](https://github.com/sablier-labs/v2-periphery/discussions/179), but his criticism at the time was in the context of the proxy target.

Since we don't have a proxy target anymore, the only solution now is to refactor the [`withdrawMultiple`](https://github.com/sablier-labs/v2-core/blob/98d5281360c9e2d4c0cac7cc8d979e8dec571de9/src/abstracts/SablierV2Lockup.sol#L328) function to implement this behavior. There are two possible approaches:

1. Remove the `to` parameter and have the smart contract query the recipient for each stream in the input array.
    - This is less gas-efficient but also less labor-intensive on the client side
2. Allow users to specify an array of `to` addresses.
    - This is more gas-efficient but also more labor-intensive on the client side
    - Technically, this approach unlocks a functionality that otherwise cannot exist, i.e, a recipient with multiple streams could call this function to withdraw to a bunch of different addresses. But this doesn't sound like a super useful feature.

There are pros and cons to each approach. Ultimately, we might want to implement both. WDYT, @razgraf, @andreivladbrg?

At a minimum, I think we should definitely implement at least the first approach, and so I have created this issue: https://github.com/sablier-labs/v2-core/issues/774

**Comment by PaulRBerg** [comment]
N.B. I've realized we are missing this feature after being asked by a potential user (Perpetual Protocol):

<img width="400" alt="SCR-20231229-obbe" src="https://github.com/sablier-labs/v2-core/assets/8782666/f3fa1a37-87fc-4e16-9ba6-ca3a484e3ad0">

**Comment by razgraf** [comment]
Resolved in #785.
